src/main/java/school/faang/user_service/service/SubscriptionService.java

```diff
@@ -0,0 +1,119 @@
+package school.faang.user_service.service;
+
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import school.faang.user_service.dto.UserDto;
+import school.faang.user_service.dto.UserFilterDto;
+import school.faang.user_service.entity.User;
+import school.faang.user_service.exception.DataValidationException;
+import school.faang.user_service.repository.SubscriptionRepository;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+@Slf4j
+public class SubscriptionService {
+    private final SubscriptionRepository subscriptionRepository;
+
+    public void followUser(long followerId, long followeeId) {
+        boolean isExistsByFollowerIdAndFolloweeId =
```

Comment:
```suggestion
        if (subscriptionRepository.existsByFollowerIdAndFolloweeId(followerId, followeeId)) {
            throw new DataValidationException("Подписка на данного пользователя уже имеется.");
        }
        subscriptionRepository.followUser(followerId, followeeId);
```
Не настаиваю, но не вижу смысла создавать тут переменнную

Reply:
Там заделка под логгер была